### PR TITLE
Add an option on POS account checkout :  #1354

### PR DESCRIPTION
**Pull request by ithiame**

### CODE REVIEW

**Comment by coyotte508** [comment]
Hmm currently the checkbox is exactly the same as "free delivery fees as POS"

Should there be a difference in behavior @Tirodem ? (Why do we need two checkboxes with the same behavior on the checkout page?)

**Comment by Tirodem** [comment]
This option should trigger the free POS delivery fees but also have an impact on billing receipt (with no delivery address being displayed)

**Comment by coyotte508** [comment]
> This option should trigger the free POS delivery fees but also have an impact on billing receipt (with no delivery address being displayed)

Yes but if both checkboxes are always synced with each other on the checkout page there's zero difference

**Comment by Tirodem** [comment]
Oh, yeah, that.
The sync must be unilateral : clicking on the new box should check the
legacy one, but seller should be able to uncheck delivery fees gratuity, to
cover some exotic functional cases

**Comment by coyotte508** [comment]
So to be clear:

- ticking new checkbox => free delivery is ticked
- unticking new checkgox => free delivery is unticked
- ticking free delivery => new checkbox doesn't change
- unticking free delivery => new checkbox doesn't change

File: src/routes/(app)/admin[[hash=admin_hash]]/config/+page.svelte [hunk 1]
```diff
@@ -165,6 +165,16 @@
 		/>
 		Limit product line per cart (for law purpose and small business )
 	</label>
+	<label class="checkbox-label">
+		<input
+			type="checkbox"
+			name="defaultOnLocation"
+			class="form-checkbox"
+			checked={data.defaultOnLocation}
```
Comment (coyotte508) [inline src/routes/(app)/admin[[hash=admin_hash]]/config/+page.svelte:173]: the "free POS delivery fees" should automatically be checked initially if this checkbox is checked

also when this checkbox changes, the other checkbox should change in consequence

**Comment by Tirodem** [comment]
> So to be clear:
> 
> * ticking new checkbox => free delivery is ticked
> * unticking new checkgox => free delivery is unticked
> * ticking free delivery => new checkbox doesn't change
> * unticking free delivery => new checkbox doesn't change

exactly ✅

**Review by coyotte508 — DISMISSED** [review]

File: src/routes/(app)/checkout/+page.server.ts [hunk 2]
```diff
@@ -335,7 +336,13 @@ export const actions = {
...
 			);
 		}
-
+		const physicalFullyPaid = z
+			.object({
+				onLocation: z.boolean({ coerce: true }).default(false)
+			})
+			.parse({
+				onLocation: formData.get('onLocation')
+			});
```
Comment (coyotte508) [inline src/routes/(app)/checkout/+page.server.ts:345]: This should only work for POS accounts

**Comment by coyotte508** [comment]
There is also the issue that `runtimeConfig.deliveryFees.allowFreeForPOS` needs to be enabled before a POS user can offer delivery fees.

How does this interact with the new `runtimeConfig.defaultOnLocation` setting @Tirodem ?

**Comment by Tirodem** [comment]
> There is also the issue that `runtimeConfig.deliveryFees.allowFreeForPOS` needs to be enabled before a POS user can offer delivery fees.
> 
> How does this interact with the new `runtimeConfig.defaultOnLocation` setting @Tirodem ?

When `runtimeConfig.deliveryFees.allowFreeForPOS` is disabled, display the option but don't seek for the free pos delivery fees option auto-tick.

Can you give me some memory refresh about `runtimeConfig.defaultOnLocation` ?

File: src/routes/(app)/checkout/+page.server.ts [hunk 3]
```diff
@@ -447,7 +457,9 @@ export const actions = {
 					...(agreements.isVATNullForeigner && {
 						acceptedExportationAndVATObligation: agreements.isVATNullForeigner
 					})
-				}
+				},
+				...(physicalFullyPaid &&
+					physicalFullyPaid.onLocation && { onLocation: physicalFullyPaid.onLocation })
```
Comment (coyotte508) [inline src/routes/(app)/checkout/+page.server.ts:462]:
```suggestion
				...(physicalFullyPaid?.onLocation && { onLocation: physicalFullyPaid.onLocation })
```

**Review by coyotte508 — APPROVED** [review]

**Comment by Tirodem** [comment]
Gosh I forgot this :
image
We need to prefill the input (I'm not a big fan) OR we need to make this field optional when the option "_Default enable "this is a fully-paid on-location physical order_"is checked, otherwise, we lose part of this feature value.

**Pull request merged**